server/app/services/openapi/v2/Swagger2SchemaGenerator.java

```diff
@@ -0,0 +1,346 @@
+package services.openapi.v2;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Locale;
+import services.applicant.question.Scalar;
+import services.openapi.OpenApiGenerationException;
+import services.openapi.OpenApiSchemaGenerator;
+import services.openapi.OpenApiSchemaSettings;
+import services.openapi.OpenApiVersion;
+import services.openapi.QuestionDefinitionNode;
+import services.openapi.v2.serializers.Swagger2YamlMapper;
+import services.program.BlockDefinition;
+import services.program.ProgramDefinition;
+import services.question.exceptions.InvalidQuestionTypeException;
+import services.question.exceptions.UnsupportedQuestionTypeException;
+import services.question.types.QuestionDefinition;
+import services.question.types.QuestionType;
+import services.question.types.ScalarType;
+
+/** Configuration used to generate a program specific swagger 2 schema */
+public class Swagger2SchemaGenerator implements OpenApiSchemaGenerator {
+  private final OpenApiSchemaSettings openApiSchemaSettings;
+
+  public Swagger2SchemaGenerator(OpenApiSchemaSettings openApiSchemaSettings) {
+    this.openApiSchemaSettings = checkNotNull(openApiSchemaSettings);
+  }
+
+  @Override
+  public String createSchema(ProgramDefinition programDefinition) {
```

Comment:
oh also, @gwendolyngoetz, I don't know how you want to structure this but the API now returns every question that has ever been part of a program, not just the questions in the current program definition.

the ProgramService as a `getAllVersionsFullProgramDefinition()` method that could be helpful, and then here we pull out all the unique question paths:
https://github.com/civiform/civiform/blob/52339de8483f1d6b75637f3a6b820ff5fee933c3/server/app/services/export/JsonExporterService.java#L124-L135

In the exporter context, we do it based on the result of `ReadOnlyApplicantProgramService::getSummaryDataAllAnswers()` cause we'll need the answers anyway and it's an easy way to get all the nested paths for enumerators. Here we don't have an application, so we'll need to do something else.

Reply:
Going to leave it as it in order to include the enum values. There's likely to be more of them than other questions that might have been removed. Once #8860 is done we can revisit.